lib/nodes/ParentNode.ts

```diff
@@ -0,0 +1,51 @@
+import {BlueshellState} from './BlueshellState';
```

Comment:
This file seems weird; I would have included the class declaration of `ParentNode` in here

Reply:
Yeah, this was done to avoid a circular import between Base and Parent.  Otherwise, everything in ParentNode would be in Parent.  See https://stackoverflow.com/questions/43176006/typeerror-class-extends-value-undefined-is-not-a-function-or-null.  I'll add a comment at the top of ParentNode.ts to explain why it's separate from Parent.